src/odb/src/lefout/lefout.cpp

```diff
@@ -144,7 +158,34 @@ void lefout::writeObstructions(dbBlock* db_block)
   dbBox* block_bounding_box = db_block->getBBox();
   for (dbTechLayer* tech_layer : obstruction_layers) {
     fprintf(_out, "    LAYER %s ;\n", tech_layer->getName().c_str());
-    writeBox("  ", block_bounding_box);
+
+    if (pin_clearance) {
+      Polygon90Set polygon_set;
+
+      for (dbBTerm* term : db_block->getBTerms()) {
+        for (dbBPin* pin : term->getBPins()) {
+          for (dbBox* box : pin->getBoxes()) {
+            if (box->getTechLayer() == tech_layer) {
+              polygon_set.insert(Rectangle(box->xMin(), box->yMin(), box->xMax(), box->yMax()));
+            }
+          }
+        }
+      }
+
+      Polygon90Set bloat_set = Polygon90Set(polygon_set + tech_layer->getSpacing());
+      Rectangle bounding_box_rect = Rectangle(block_bounding_box->xMin(), block_bounding_box->yMin(),
+                                               block_bounding_box->xMax(), block_bounding_box->yMax());
+      Polygon90Set obs_set = Polygon90Set(bounding_box_rect - bloat_set);
+
+      std::vector<Rectangle> rects;
+      obs_set.get_rectangles(rects);
+
+      for (auto& r : rects) {
+        writeBox("  ", xl(r), yl(r), xh(r), yh(r));
+      }
```

Comment:
I think this code is great it's definitely more correct than what was here before, but will this actually impact the router?

From my understanding the router will override obstructions if the macro has pins on edge of the macro, and if the pin is on the inside of the macro it needs to accessible from either around the the pin or from above or below. In this case we're removing the obstructions around pins, but not from layers above or below.

Reply:
> I think this code is great it's definitely more correct than what was here before, but will this actually impact the router?
> 
> From my understanding the router will override obstructions if the macro has pins on edge of the macro, and if the pin is on the inside of the macro it needs to accessible from either around the the pin or from above or below. In this case we're removing the obstructions around pins, but not from layers above or below.

Does grt and drt handle this? I made up a test case with a pin on the edge of a macro with an overlapping obstruction, and it failed in drt pin access.

Reply:
I think what you have here is good, but if it doesn't work I believe that's a bug. @maliberty can you confirm.

Reply:
The comments starting with "Min space is not necessarily correct" have not been addressed yet so it is not ready imo.

Reply:
Sorry didn't mean to imply that this specific code should be merged, but the technique of removing obstructions around pins is good. I was asking you to confirm if Anton's case was in fact a bug.

Reply:
I need to see the test case to see what the issue is. It should work.

Reply:
I'm not sure if it was recent updates to the pin access code, or just bugs in my test cases, but I've now tested three variants (macro sized obstruction, obstruction cut around pins with no clearance, clearance around pins). All work.